### PR TITLE
feat(iothub-service): Expose DTDL model Id property for pnp devices.

### PR DESCRIPTION
**Pull request by vinagesh**
## Description of the changes
Adding changes to serialize and de-serialize DTDL modelId property returned by the Service on GET /twins/<deviceId> calls.

### CODE REVIEW

File: shared/src/TwinJsonConverter.cs [hunk 1]
```diff
@@ -217,6 +224,10 @@ public override object ReadJson(JsonReader reader, Type objectType, object exist
                         twin.DeviceId = reader.Value as string;
                         break;
 
+                    case ModelId:
```
Comment (vinagesh) [inline shared/src/TwinJsonConverter.cs:227]: This is to deserialize

File: shared/src/TwinJsonConverter.cs [hunk 2]
```diff
@@ -72,10 +73,16 @@ public override void WriteJson(JsonWriter writer, object value, JsonSerializer s
...
+            {
+                writer.WritePropertyName(ModelId);
+                writer.WriteValue(twin.ModelId);
+            }
+
             writer.WritePropertyName(DeviceIdJsonTag);
             writer.WriteValue(twin.DeviceId);
 
-            if (!string.IsNullOrEmpty(twin.ModuleId))
+            if (!string.IsNullOrWhiteSpace(twin.ModuleId))
```
Comment (vinagesh) [inline shared/src/TwinJsonConverter.cs:85]: This is to serialize

File: shared/src/TwinJsonConverter.cs [hunk 3]
```diff
@@ -39,6 +39,7 @@ public sealed class TwinJsonConverter : JsonConverter
         private const string CloudToDeviceMessageCountTag = "cloudToDeviceMessageCount";
         private const string AuthenticationTypeTag = "authenticationType";
         private const string X509ThumbprintTag = "x509Thumbprint";
+        private const string ModelId = "digital-twin-model-id";
```
Comment (prmathur-microsoft) [inline shared/src/TwinJsonConverter.cs:43]: Is this change working now? This is expected to change fyi.

File: iothub/service/src/ClientApiVersionHelper.cs [hunk 4]
```diff
@@ -12,7 +12,7 @@ internal class ClientApiVersionHelper
     {
         private const string ApiVersionQueryPrefix = "api-version=";
         private const string ApiVersionGA = "2016-02-03";
-        private const string ApiVersionDefault = "2018-08-30-preview";
+        private const string ApiVersionDefault = "2020-05-31-preview";
         private const string ApiVersionLimitedAvailability = "2020-03-13";
```
Comment (prmathur-microsoft) [inline iothub/service/src/ClientApiVersionHelper.cs:16]: How are you verifying this works e2e?

File: shared/src/Twin.cs [hunk 5]
```diff
@@ -48,6 +48,13 @@ public Twin(TwinProperties twinProperties)
         /// </summary>
         public string DeviceId { get; set; }
 
+        /// <summary>
+        /// The DTDL model id of the device.
+        /// The value will be null for a non-pnp device.
+        /// The value will be null for a pnp device until the device connects and registers with the model id.
+        /// </summary>
+        public string ModelId { get; set; }
+
```
Comment (prmathur-microsoft) [inline shared/src/Twin.cs:57]: Is there an e2e possible for this scenario?

**Comment by abhipsaMisra** [comment]
From Miguel's email today to PnPPartners alias, it looks like this is now available on East US 2 EUAP; we should be able to add a test around this

File: shared/src/TwinJsonConverter.cs (hunk 3 above)
Comment (abhipsaMisra) [reply shared/src/TwinJsonConverter.cs:43]: Isn't the change specific to what we send on the mqtt connect packet? This shouldn't be affected in that case.
Or are we talking about a different change here?

File: shared/src/Twin.cs (hunk 5 above)
Comment (timtay-microsoft) [reply shared/src/Twin.cs:57]: It'll depend on if the hub's that support this are available in our canary region or not.

File: shared/src/TwinJsonConverter.cs (hunk 3 above)
Comment (vinagesh) [reply shared/src/TwinJsonConverter.cs:43]: Yes it is working. 1) There is backward compatibility as existing tests are passing 2) This has been manually tested using samples against the hub they provided

File: shared/src/Twin.cs (hunk 5 above)
Comment (vinagesh) [reply shared/src/Twin.cs:57]: Not yet. There is a task on my card to do it when their service is deployed in Canary.

Comment (vinagesh) [reply shared/src/Twin.cs:57]: Based on Abhipsa's update below the service might be available today. So i'll add it in the next PR.

File: iothub/service/src/ClientApiVersionHelper.cs (hunk 4 above)
Comment (vinagesh) [reply iothub/service/src/ClientApiVersionHelper.cs:16]: The E2E is running in the pipeline so it works in terms of existing scenarios. For new scenarios, as I mentioned in all other places, a test will be added when the service is in Canary.

File: shared/src/TwinJsonConverter.cs (hunk 3 above)
Comment (prmathur-microsoft) [reply shared/src/TwinJsonConverter.cs:43]: Great that it is working. Then lets take the followup change in next PR. The rename applies to both device and service SDK's.

**Comment by bikamani** [comment]
Merging this change as per offline discussion with @vinagesh

**Pull request merged**